rke/k8s_defaults.go

```diff
@@ -0,0 +1,103 @@
+package rke
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/rancher/kontainer-driver-metadata/rke/templates"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/image"
+)
+
+const (
+	rkeDataFilePath = "./data/data.json"
+)
+
+// Data to be written in dataFilePath, dynamically populated on init() with the latest versions
+type Data struct {
+
+	// K8sVersionServiceOptions - service options per k8s version
+	K8sVersionServiceOptions  map[string]v3.KubernetesServicesOptions
+	K8sVersionRKESystemImages map[string]v3.RKESystemImages
+
+	// Addon Templates per K8s version ("default" where nothing changes for k8s version)
+	K8sVersionedTemplates map[string]map[string]string
+
+	// K8sVersionToRKEVersions - min/max RKE versions per k8s version
+	K8sVersionToRKEVersions map[string]v3.RKEVersion
+
+	// K8sVersionToRancherVersions - min/max Rancher versions per k8s version
+	K8sVersionToRancherVersions map[string]v3.RancherVersion
+
+	//Default K8s version for every rancher version
+	RancherDefaultK8sVersions map[string]string
+
+	//Default K8s version for every rke version
+	RKEDefaultK8sVersions map[string]string
+
+	K8sVersionWindowsSystemImages   map[string]v3.WindowsSystemImages
+	K8sVersionWindowsServiceOptions map[string]v3.KubernetesServicesOptions
+}
+
+var (
+	DriverData Data
+	m          = image.Mirror
+)
+
+func init() {
+	DriverData = Data{
+		K8sVersionRKESystemImages: loadK8sRKESystemImages(),
+	}
+
+	for version, images := range DriverData.K8sVersionRKESystemImages {
+		longName := "rancher/hyperkube:" + version
+		if !strings.HasPrefix(longName, images.Kubernetes) {
```

Comment:
is this mostly to protect from user (admin) error? when the image he/she enters, is incorrect

Reply:
yeah, followed the same logic as we do in types